donkeycar/parts/path.py

```diff
@@ -385,11 +432,21 @@ def run(self, path, x, y, from_pt=None):
 
 class PID_Pilot(object):
 
-    def __init__(self, pid, throttle):
+    def __init__(
+            self,
+            pid: PIDController,
+            throttle: float,
+            use_constant_throttle: bool = False) -> None:
         self.pid = pid
         self.throttle = throttle
+        self.use_constant_throttle = use_constant_throttle
+        self.variable_speed_multiplier = 1.0
 
-    def run(self, cte):
+    def run(self, cte: float, throttles: list, closest_pt_idx: int) -> tuple:
         steer = self.pid.run(cte)
-        logging.info("CTE: %f steer: %f" % (cte, steer))
-        return steer, self.throttle
+        if self.use_constant_throttle or throttles is None or closest_pt_idx is None:
```

Comment:
This line makes me think there is a better way to handle minimum throttle than to record a minimum throttle into the path.  What if we allow ourselves to record the real throttle always, but here we enforce the minimum throttle;
```
if self.use_constant_throttle or throttles is None or closest_pt_idx is None or throttles[closest_pt_idx] < self.throttle:
```
So if the recorded throttle is less than the minimum then we use the minimum.  This had the big advantage that 1) that data has the actual throttle, 2) we can change the minimum without altering the data.

What do you think?

Reply:
I think that sounds like a great idea 👍🏼